pkg/eventmonitor/eventmonitor.go

```diff
@@ -156,6 +161,12 @@ func (m *EventMonitor) Start() error {
 		return err
 	}
 
+	for _, em := range m.eventConsumers {
+		if err := em.PostStart(); err != nil {
+			log.Errorf("after probe start callback of %s failed: %v", em.ID(), err)
```

Comment:
Should this happen before line 154 where all the consumers start started? The name here implies to me that there is a guarantee that PostStart runs before any events are sent, which isn't strictly guaranteed by this order. 

In that vein, would it make sense to stop or never start a given consumer if it's `PostStart` returns an error?

Reply:
The goal here is just to have a callback after the event stream started to flow. We cannot do it line 154 because the probe has not started yet, which the entire goal of this PR